src/object_accessor.hpp

```diff
@@ -128,91 +94,75 @@ struct MutationOutsideTransactionException : public std::logic_error {
 template <typename ValueType, typename ContextType>
 void Object::set_property_value(ContextType ctx, std::string prop_name, ValueType value, bool try_update)
 {
-    const Property *prop = m_object_schema->property_for_name(prop_name);
-    if (!prop) {
-        throw InvalidPropertyException(m_object_schema->name, prop_name,
-            "Setting invalid property '" + prop_name + "' on object '" + m_object_schema->name + "'.");
-    }
-    set_property_value_impl(ctx, *prop, value, try_update);
+    verify_attached();
+    m_realm->verify_in_write();
+    auto& property = property_for_name(prop_name);
+    if (property.is_primary)
+        throw std::logic_error("Cannot modify primary key after creation");
+
+    set_property_value_impl(ctx, property, value, try_update);
 }
 
 template <typename ValueType, typename ContextType>
 ValueType Object::get_property_value(ContextType ctx, std::string prop_name)
 {
-    const Property *prop = m_object_schema->property_for_name(prop_name);
-    if (!prop) {
-        throw InvalidPropertyException(m_object_schema->name, prop_name,
-            "Getting invalid property '" + prop_name + "' on object '" + m_object_schema->name + "'.");
-    }
-    return get_property_value_impl<ValueType>(ctx, *prop);
+    return get_property_value_impl<ValueType>(ctx, property_for_name(prop_name));
 }
 
 template <typename ValueType, typename ContextType>
-void Object::set_property_value_impl(ContextType ctx, const Property &property, ValueType value, bool try_update)
+void Object::set_property_value_impl(ContextType ctx, const Property &property, ValueType value, bool try_update, bool is_default)
 {
     using Accessor = NativeAccessor<ValueType, ContextType>;
 
-    verify_attached();
-
-    if (!m_realm->is_in_transaction()) {
-        throw MutationOutsideTransactionException("Can only set property values within a transaction.");
-    }
-
+    auto& table = *m_row.get_table();
     size_t column = property.table_column;
+    size_t row = m_row.get_index();
     if (property.is_nullable && Accessor::is_null(ctx, value)) {
```

Comment:
The code currently expects `NativeAccessor` to handle checking for type mismatches, including throwing an appropriate error when `Accessor::to_foo()` is called on a null value.